chess-AI/train.py

```diff
@@ -100,7 +103,7 @@ def assign_rewards(self, board, length):
         print(board.outcome())
         return values
 
-    def training_episode(self, nnet, games, epochs, batch_size):
+    def training_episode(self, nnet, games, epochs, batch_size, num_saved_models, overwrite_save):
```

Comment:
I think we're getting to the point where we should define a `TrainOptions` class that we can pass a parameter to this function to make the function signature more clear and make it easier to add parameters later without modifying the function signature. Maybe we can work on that in a new PR after addressing all the other comments!
That way the function signature will always be something like
`def training_episode(self, nnet, train_ops):`
and we can have all the things currently being passed to the function as instance variables of the class.